From 289ccf8f0e77ae4f240eca988eb3a6ac19afbae9 Mon Sep 17 00:00:00 2001
From: Thomas Bonnefille <thomas.bonnefille@bootlin.com>
Date: Fri, 17 Jan 2025 22:10:41 +0100
Subject: [PATCH] Replace cuserid

cuserid has been removed from some toolchains and is no longer in the
POSIX standard since 1990. getlogin has been in the standard since 2001.

This commit uncomments the call to getlogin and thus replaces cuserid
with getlogin.

Upstream: https://github.com/torvalds/uemacs/pull/46
Signed-off-by: Thomas Bonnefille <thomas.bonnefille@bootlin.com>
---
 pklock.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/pklock.c b/pklock.c
index dc30b76..5f4865b 100644
--- a/pklock.c
+++ b/pklock.c
@@ -80,8 +80,7 @@ char *dolock(char *fname)
 	}
 	if ((n = read(fd, locker, MAXNAME)) < 1) {
 		lseek(fd, 0, SEEK_SET);
-/*		strcpy(locker, getlogin()); */
-		cuserid(locker);
+		strcpy(locker, getlogin());
 		strcat(locker + strlen(locker), "@");
 		gethostname(locker + strlen(locker), 64);
 		write(fd, locker, strlen(locker));
-- 
2.48.1

